client/me/billing-history/utils.js

```diff
@@ -9,7 +9,10 @@ import { find, map, partition, reduce } from 'lodash';
  */
 import formatCurrency from 'lib/format-currency';
 
-export const groupDomainProducts = ( transactionItems, translate ) => {
+export const groupDomainProducts = ( originalItems, translate ) => {
+	const transactionItems = Object.keys( originalItems ).map( key => {
+		return Object.assign( {}, originalItems[ key ] );
+	} );
```

Comment:
Good catch, if we basically need a deep clone here, I'd suggest using lodash [cloneDeep](https://lodash.com/docs/4.17.10#cloneDeep) Since we use this library.

Reply:
We could; it's just one layer so I'm fine with a map, but if you think that `cloneDeep` would be faster/more efficient/more clear, I'm happy to do that instead.

Reply:
Non-blocking, and perf for calling lodash vs this shouldn't be an issue. The intent is a bit clearer imo with the suggestion.